code/scripting/api/libs/tables.cpp

```diff
@@ -218,6 +219,27 @@ ADE_FUNC(__len, l_Tables_FireballClasses, NULL, "Number of fireball classes", "n
 
 	return ade_set_args(L, "i", Num_fireball_types);
 }
+
+ADE_LIB_DERIV(l_Tables_SimulatedSpeechOverrides, "SimulatedSpeechOverrides", NULL, NULL, l_Tables);
+
+ADE_INDEXER(l_Tables_SimulatedSpeechOverrides, "number Index", nullptr, "string", "Truncated filenames of simulated speech overrides or empty string if index is out of range.")
+{
+	int idx = -1;
+	if (!ade_get_args(L, "*i", &idx)) {
+		return ade_set_error(L, "s", "");
+	}
+	if (idx > 0 && (unsigned int) idx <= Generic_message_filenames.size()) {
```

Comment:
You need to use `size_t` here to get the correct type for either 32- or 64-bit:
```suggestion
	if (idx > 0 && (size_t) idx <= Generic_message_filenames.size()) {
```